packages/@aws-cdk/aws-codepipeline-actions/test/integ.pipeline-ecs-create-cross-account.lit.ts

```diff
@@ -0,0 +1,221 @@
+        actions: ['iam:PassRole'],
+        resources: ['*'],
+        conditions: {
+          StringEqualsIfExists: {
+            'iam:PassedToService': [
+              'ec2.amazonaws.com',
+              'ecs-tasks.amazonaws.com',
+            ],
+          },
+        },
+      }),
+    );
+
+  }
+}
+
+/**
+ * This is the Stage which does our create using {@link TestCreateEcsStack}.
+ */
+class TestCreateEcsStage extends cdk.Stage {
+  public readonly serviceArn: string;
+  public readonly clusterName: string;
+  public readonly deployRole: iam.IRole;
+
+  constructor(scope: Construct, id: string, props: cdk.StageProps) {
+    super(scope, id, props);
+    const testStack = new TestCreateEcsStack(this, 'ecsStack', {});
+    this.serviceArn = testStack.serviceArn;
+    this.clusterName = testStack.clusterName;
+    this.deployRole = testStack.deployRole;
+  }
+}
+
+/**
+ * This is our pipeline which will create an ECS Service and deploy
+ * to is using {@link EcsDeployAction} using our {@link TestCreateEcsStage}
+ */
+class PipelineEcsDeployCreateStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props: cdk.StackProps) {
+    super(scope, id, props);
+    const artifact = new codepipeline.Artifact('Artifact');
+    const bucket = new s3.Bucket(this, 'PipelineBucket', {
+      versioned: true,
+      removalPolicy: cdk.RemovalPolicy.DESTROY,
+    });
+    const source = new cpactions.S3SourceAction({
+      actionName: 'Source',
+      output: artifact,
+      bucket,
+      bucketKey: 'key',
+    });
+    const pipeline = new codepipeline.Pipeline(this, 'Pipeline', {
+      stages: [
+        {
+          stageName: 'Source',
+          actions: [source],
+        },
+      ],
+    });
+    const testStage = new TestCreateEcsStage(
+      this,
+      'TestStage',
+      {
+        env: {
+          region: 'us-west-2',
+          account: '123456789012',
+        },
+      },
+    );
+    const testIStage = pipeline.addStage(testStage);
+    /**
+     * This imports the service so it can be used by the ECS Deploy Action.
+     * This must use the serviceArn so that the region is set correctly.
+     * The VPC doesn't actually matter and it doesn't get created.
+     * The construct ids will need to be unique.
+     */
+    const service = ecs.FargateService.fromFargateServiceAttributes(this, 'FargateService', {
+      serviceArn: testStage.serviceArn,
+      cluster: ecs.Cluster.fromClusterAttributes(this, 'Cluster', {
+        vpc: new ec2.Vpc(this, 'Vpc'),
+        securityGroups: [],
+        clusterName: testStage.clusterName,
+      }),
+    });
+    /**
+     * It is highly recommended passing in the role from the stage/stack, if not a new role
+     * will be added to the pipeline action, will not exist in the account you are deploying to.
+     */
+    const deployAction = new cpactions.EcsDeployAction({
+      actionName: 'ECS',
+      service: service,
+      imageFile: artifact.atPath('imageFile.json'),
```

Comment:
I have to say, I don't understand where is this `imageFile.json` file coming from. Is it part of the source of your application? Should it be the result of some build that you do?

Reply:
yes it could be the result of the build, in this case I am point it to the source code repo, so imageFile.json would be in the code commit repo.  I could change to just us the import for the example if you prefer.

Reply:
Let's maybe just use `input` instead of `imageFile` in `EcsDeployAction`, and leave a comment in the code here.

Reply:
updated and add a comment.